test/wallet-http-test.js

```diff
@@ -1320,6 +1321,152 @@ describe('Wallet HTTP', function() {
     const mempool = await nclient.getMempool();
     assert.ok(mempool.length === 0);
   });
+
+  it('should create a batch reveal transaction (multiple outputs) for partial valid names', async function() {
+    const VALID_NAMES_LEN = 2;
+    const validNames = [];
+    for (let i = 0; i < VALID_NAMES_LEN; i++) {
+      validNames.push(await nclient.execute('grindname', [5]));
+    }
+    const INVALID_NAMES_LEN = 10;
+    const invalidNames = [...Array(INVALID_NAMES_LEN).keys()];
+
+    await mineBlocks(1, cbAddress);
+
+    await wclient.createBatchOpen('primary', {
+      names: validNames,
+      passphrase: '',
+      broadcast: true,
+      sign: true
+    });
+
+    await mineBlocks(treeInterval + 1, cbAddress);
+
+    // TODO Promise.All is failing ?
+    const numberOfBids = VALID_NAMES_LEN * 2;
+    for (const domainName of validNames) {
+      await wallet.createBid({
+        name: domainName,
+        bid: 1000,
+        lockup: 2000
+      });
+      await wallet.createBid({
+        name: domainName,
+        bid: 1500,
+        lockup: 2000
+      });
+    }
+
+    await mineBlocks(biddingPeriod + 1, cbAddress);
+
+    const json = await wclient.createBatchReveal('primary', {
+      passphrase: '',
+      names: [...validNames, ...invalidNames],
+      sign: true,
+      broadcast: true
+    });
+
+    const transaction = json['tx'];
```

Comment:
style nit: can do `const { tx: transaction, errors } = json;`

Reply:
great tip will keep in mind ... fixed ...